### PR TITLE
Fix duplicate, unclickable row when re-dropping the same track

### LINKED ISSUE #2: When the same track has been separated more than once, the UI behaves erratically
If I separated the same track more than once, the UI for displaying the separated tracks shows duplicate rows and the hover/click behaviors appear broken. I can only interact with the oldest item in the list of duplicate rows. (PR coming)

### PR DESCRIPTION
**Pull request by shyam-habarakada**
Fixes nsaintot/stemd#2

## The bug

If you drop the same track on the window twice (or drop it again after it already finished separating), the "Separated" list ended up with two rows for it instead of one. The newer of the two rows was broken: hovering and clicking did nothing on it. To get any hover/click feedback on that track, you had to move the mouse onto the *older* duplicate row further down the list instead.

## Why it happened

Two separate things combined to cause this:

1. **Nothing de-duplicated the list.** Every time a file was dropped, the window just inserted a brand-new row at the top of the "recent drops" list, with no check for whether that same file was already sitting somewhere in the list. So re-running a track always added a second row rather than updating the one already there.

2. **Rows were sharing an identity behind the scenes.** Each row's click/hover handling needs a stable ID so the UI toolkit (egui) can tell "this row was pressed" from "that row was pressed." That ID was built from the track's display name (the file's title). But two rows for the same file obviously have the same title, so both rows ended up using the *exact same ID*. egui only routes input correctly when every interactive element has a unique ID, so with two rows sharing one ID, input got misattributed — the newer row looked normal but silently couldn't be clicked, while the older row (usually hidden further down the list) picked up the intended interaction instead.

## The fix

- When a track is dropped, the window now first removes any existing row for that same source file before adding the new one, then puts the new row at the top. So re-dropping a track now moves it to the top of the list and refreshes its status, instead of piling up a duplicate.
- Row IDs are now built from the file's full path instead of its display name. This is a more accurate identity anyway (two different files in different folders can share the same name), and it means two rows can never again collide on the same ID, so this class of bug can't resurface even in that edge case.

## Tests added

Added a test that simulates dropping the same file twice: it drops file A, then drops an unrelated file B, then drops file A again, and checks that:
- the list still has exactly two rows (no duplicate was created),
- the re-dropped file A is now at the top of the list,
- the unrelated file B's row wasn't disturbed.

All existing tests continue to pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/stemd-server/src/drops.rs
Comment (nsaintot) [inline crates/stemd-server/src/drops.rs:168]: Comment is stale with the dedup. Trim or update

File: crates/stemd-server/src/drops.rs [hunk 1]
```diff
@@ -127,11 +140,7 @@ impl Drops {
             state: Mutex::new(State::Reading),
         });
 
-        {
-            let mut recent = self.recent.lock();
-            recent.insert(0, Arc::clone(&dropped));
-            recent.truncate(REMEMBERED);
-        }
+        remember(&mut self.recent.lock(), Arc::clone(&dropped));
```
Comment (nsaintot) [inline crates/stemd-server/src/drops.rs:143]: [Nit] Dedup lives only in accept; nothing structural stops a future insert from bypassing remember.

I would tend to favor a method for `remember` rather than a free function that only has one caller.

**Review by nsaintot — CHANGES_REQUESTED** [review]
Overall good QoL PR

**Comment by nsaintot** [comment]
dup of #4
